### PR TITLE
Use private DNS record when not associating public IP

### PR DESCRIPTION
**Pull request by tyronedd**
## What
* Use private DNS record when `var.associate_public_address = false`.

## Why
* When not associating public IP the record would be empty. This fixes that.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -110,5 +110,5 @@ module "dns" {
   name    = module.this.name
   zone_id = var.zone_id
   ttl     = 60
-  records = aws_instance.default.*.public_dns
+  records = var.associate_public_address ? aws_instance.default.*.public_dns : aws_instance.default.*.private_dns
```
Comment (joe-niland) [inline main.tf:113]: variable name is associate_public_ip_address

**Comment by joe-niland** [comment]
@tyronedd btw if you want to run tests locally:

```
cd test/src
make test
# or
make docker/test
```

File: main.tf (hunk 1 above)
Comment (tyronedd) [reply main.tf]: Ahh, stupid, used the online editor. I think it should be correct now.

I executed the commands below, but it failed on AWS creds.

**Review by joe-niland — APPROVED** [review]

**Comment by joe-niland** [comment]
@tyronedd thanks for your contribution!

**Pull request merged**